x-pack/auditbeat/module/system/socket/socket_linux.go

```diff
@@ -0,0 +1,361 @@
+	"github.com/elastic/beats/x-pack/auditbeat/module/system"
+	"github.com/elastic/beats/x-pack/auditbeat/module/system/socket/guess"
+	"github.com/elastic/beats/x-pack/auditbeat/module/system/socket/helper"
+	"github.com/elastic/beats/x-pack/auditbeat/tracing"
+	"github.com/elastic/go-perf"
+	"github.com/elastic/go-sysinfo"
+	"github.com/elastic/go-sysinfo/providers/linux"
+)
+
+const (
+	moduleName     = "system"
+	metricsetName  = "socket"
+	fullName       = moduleName + "/" + metricsetName
+	namespace      = "system.audit.socket"
+	detailSelector = metricsetName + "detailed"
+	auditbeatGroup = "auditbeat"
+	// Magic value to detect clock-sync events generated by the metricset.
+	clockSyncMagic uint64 = 0x42DEADBEEFABCDEF
+)
+
+var (
+	kernelVersion string
+	eventCount    uint64
+)
+
+// MetricSet for system/socket.
+type MetricSet struct {
+	system.SystemMetricSet
+	templateVars common.MapStr
+	config       Config
+	log          *logp.Logger
+	detailLog    *logp.Logger
+	installer    helper.ProbeInstaller
+	perfChannel  *tracing.PerfChannel
+	isDebug      bool
+	isDetailed   bool
+}
+
+func init() {
+	mb.Registry.MustAddMetricSet(moduleName, metricsetName, New,
+		mb.DefaultMetricSet(),
+		mb.WithNamespace(namespace),
+	)
+	var err error
+	if kernelVersion, err = linux.KernelVersion(); err != nil {
+		logp.Err("Failed fetching Linux kernel version: %v", err)
+	}
+}
+
+// New constructs a new MetricSet.
+func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
+	cfgwarn.Beta("The %s dataset is beta.", fullName)
+
+	config := defaultConfig
+	if err := base.Module().UnpackConfig(&config); err != nil {
+		return nil, errors.Wrapf(err, "failed to unpack the %s config", fullName)
+	}
+
+	ms := &MetricSet{
+		SystemMetricSet: system.NewSystemMetricSet(base),
+		templateVars:    make(common.MapStr),
+		config:          config,
+		log:             logp.NewLogger(metricsetName),
+		isDebug:         logp.IsDebug(metricsetName),
+		detailLog:       logp.NewLogger(detailSelector),
+		isDetailed:      logp.IsDebug(detailSelector),
+	}
+
+	// Setup the metricset before Run() so that startup can be halted in case of
+	// error.
+	if err := ms.Setup(); err != nil {
+		return nil, errors.Wrapf(err, "%s dataset setup failed", fullName)
+	}
+	return ms, nil
+}
+
+// Run the metricset. This will loop until the passed reporter is cancelled.
+func (m *MetricSet) Run(r mb.PushReporterV2) {
+	defer m.log.Infof("%s terminated.", fullName)
+	defer m.Cleanup()
+
+	if err := m.perfChannel.Run(); err != nil {
+		err = errors.Wrap(err, "unable to start perf channel")
+		r.Error(err)
+		m.log.Error(err)
+		return
+	}
+	// Launch the clock-synchronization ticker.
+	go m.clockSyncLoop(m.config.ClockSyncPeriod, r.Done())
+	st := NewState(r,
+		m.log,
+		m.config.FlowInactiveTimeout,
+		m.config.FlowTerminationTimeout,
+		m.config.ClockMaxDrift)
+
+	if procs, err := sysinfo.Processes(); err != nil {
+		m.log.Error("Failed to bootstrap process table using /proc", err)
```

Comment:
Should this be considered fatal? or the idea is to keep it working even if `/proc` is not available?

Reply:
Yes, this is only to populate state with existing processes. The only side-effect of this failing will be that information about existing processes is not added to events.